internal/catalog/internal/types/errors.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package types
```

Comment:
These error helpers are great! I wonder if we'll want to promote some of them to the `internal/resource` package?

Reply:
I moved these to a new internal/resource/utils package (at least the ones which would make sense outside of the catalog)